app/javascript/utils/DelegationRequests.js

```diff
@@ -43,16 +45,27 @@ export async function delegateQuestions(answers, buildingId, email, firstName, l
     });
     addAnswers(answersToUpdate, buildingId);
   } catch (error) {}
+
+  if (callbackFn) {
+    callbackFn();
+  }
 }
 
-export async function delegateBuildingQuestions(buildingId, userDetails, state, addAnswers) {
+export async function delegateBuildingQuestions(buildingId, userDetails, state, addAnswers, callbackFn=null) {
   let answers = getUnfinishedAnswersForBuilding(buildingId, state);
 
   await delegateQuestions(answers, buildingId, userDetails.email, userDetails.firstName, userDetails.lastName, addAnswers);
+
+  if (callbackFn) {
+    callbackFn()
+  }
 }
 
-export async function delegateCategoryQuestions(categoryId, buildingId, userDetails, state, addAnswers) {
+export async function delegateCategoryQuestions(categoryId, buildingId, userDetails, state, addAnswers, callbackFn=null) {
   let answers = getUnfinishedAnswersForCategoryAndBuilding(categoryId, buildingId, state);
 
   await delegateQuestions(answers, buildingId, userDetails.email, userDetails.firstName, userDetails.lastName, addAnswers);
+  if (callbackFn) {
```

Comment:
[for the future]: I feel like there's a better way to handle having a callback function so we don't need to have lots of ifs to check if one is passed on